pkgs/native_toolchain_c/lib/src/native_toolchain/msvc.dart

```diff
@@ -63,9 +63,15 @@ Tool vcvars(ToolInstance toolInstance) {
   final tool = toolInstance.tool;
   assert(tool == cl || tool == link || tool == lib);
   final vcDir = toolInstance.uri.resolve('../../../../../../');
-  final fileName = toolInstance.uri.toFilePath().contains('x86')
-      ? 'vcvars32.bat'
-      : 'vcvars64.bat';
+  var fileName = 'vcvars64.bat';
```

Comment:
Prefer:

```dart
final fileName;
if(...)
...
```

This enforces exactly one assignment to the variable with control flow analysis, and afterwards assignments are not allowed.

Reply:
Nice, I was looking exactly for this, but didn't find this construction.